src/mlpack/core/data/dictionary_encoding.hpp

```diff
@@ -0,0 +1,142 @@
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  *
+  */
+  template<typename TokenizerType>
+  void CreateMap(std::string& strings, TokenizerType tokenizer);
+  /*
+  * Default Constructor
+  */
+  DicitonaryEncoding() {}
+  /*
+  * Copy Constructor
+  */
+  DicitonaryEncoding(const DicitonaryEncoding &old_obj);
+  /*
+  * Assignment Operators
+  */
+  void operator= (const DicitonaryEncoding &old_obj);
+  /**
+  * A function to reset the mapping that is clear all the encodings
+  */
+  void Reset();
+
+  /**
+  * A fucntion to encode given array of strings using a particular delimiter,
+  * providing custom rule for tokenization.
+  *
+  * For example 
+  * Vector is :
+  * [hello@wow, wow@hello@good] would be encoded using '@' as delimiter as 
+  * [1 2 0, 2 1 3] 
+  * The function paddes 0 to maintain same sizes across all the rows
+  * User may also provide their custom tokenization rule.
+  *
+  * @param strings Vector of strings.
+  * @param output Output Matrix to store encoded results (sp_mat or mat).
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  *
+  */
+  template<typename MatType, typename TokenizerType>
+  void Encode(const std::vector<std::string>& strings,
+                  MatType& output, TokenizerType tokenizer);
+
+  /**
+  * A function to encode given array of strings using a particular delimiter,
+  * with custome tokenization.
+  *
+  * For example 
+  * Vector is :
+  * [hello@wow, wow@hello@good] would be encoded using '@' as delimiter as 
+  * [1 2 , 2 1 3] 
+  * The function does not paddes 0 in this case.
+  *
+  * @param strings Vector of strings.
+  * @param output Vector of vectors to store encoded results.
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  *
+  */
+  template<typename TokenizerType>
+  void Encode(const std::vector<std::string>& strings,
+            std::vector<std::vector<size_t>>& output, TokenizerType tokenizer);
+
+  //! Modify the stringq.
+  std::deque<std::string>& Stringq() { return stringq; }
+
+  //! Return the Mappings
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& Mappings() const { return mappings; }
+
+  //! Modify the Mappings.
+  std::tuple<std::unordered_map<boost::string_view, size_t,
```

Comment:
Sorry, I think i was wrong. If you provide a method for modification `stringq` then it is reasonable to provide a method for modification `mappings`. I suggest the following code:
```
std::deque<std::string>& Stringq() { return stringq; }
const std::deque<std::string>& Stringq() const { return stringq; }
const std::unordered_map<boost::string_view, size_t,
                                            boost::hash<boost::string_view>>& Mappings() cosnt { return mappings; }
std::unordered_map<boost::string_view, size_t,
                                  boost::hash<boost::string_view>>& Mappings() { return mappings; }
```

Reply:
Reverted back to old interface :)